### ISSUE TITLE
Inconsistency in TableWaveform

### ISSUE THREAD

**Issue by terrorfisch** [qutech/qupulse#475]
`TableWaveform.duration` is `TimeType` but the table entries are  of type `float`

**Comment by terrorfisch** [comment]
The other possible fix besides #476 is to switch the internal representation to `TimeType`. I suspect that this will decrease the performance.

**Comment by peendebak** [comment]
@terrorfisch In the following example the type in the tables entries is not `float`, but `Expression`
```
import numpy as np
from qupulse.expressions import Expression
from qupulse.pulses import TablePT
from qupulse.pulses.plotting import render

v=Expression('-0.0158904109589041')
template2=TablePT({'P1': [(0,v, 'hold'), (Expression(3999.9999999999886),v, 'hold')]})
sample_rate=.1    

for ii in range(10):
    r=render(template2.create_program())
    data=r[1]['P1'] 
    print(data[-1])
```
Maybe we can restrict the entries of the `TablePT` to be `float`?

**Comment by terrorfisch** [comment]
The type is always expression until the pulse template gets instantiated. Then it is evaluated to an "allowed type" which is asserted in `Expression._parse_evaluate_numeric_result`.

**Comment by terrorfisch** [comment]
Fixed in #476
